lib/connection.js

```diff
@@ -1,12 +1,126 @@
 'use strict';
 
 const common = require('metarhia-common');
-const EventEmitter = require('events');
+const { EventEmitter } = require('events');
 
-const Connection = function() {
+const parser = require('./parser.js');
+
+const Connection = function(transport) {
   EventEmitter.call(this);
+
+  this._transport = transport;
```

Comment:
`Client` and `ClientConnection` will extent `Connection` and it will add `this.transport` to certain implementation. This is abstract class and it's constructor should have identical contract with `Client` and `ClientConnection`.